tests/test_utils.py

```diff
@@ -764,6 +764,13 @@ def test_status(self, app):
         result = client.simulate_get()
         assert result.status == falcon.HTTP_702
 
+    def test_repr_result(self, app):
```

Comment:
IMHO it would be nice to add more tests both for WSGI and ASGI flavours of the framework, potentially using `simulate_request()` and friends.